`MoveResourceState` fails with `attributes "id" and "type" are required` regression starting in v2.8.0

### Description

State migration via `moved {}` blocks (moving from an `azurerm` resource like `azurerm_resource_group` to an `azapi_resource`) fails during the planning phase (`terraform plan` / `tofu plan`) with the following error:

```text
Error: attributes "id" and "type" are required
```

This regression affects all provider versions starting from **`v2.8.0`** (when `IdentitySchema` was first introduced) up to the latest **`v2.11.0`**. The state migration works perfectly in version **`v2.7.0`** (prior to `IdentitySchema`).

---

### Root Cause Analysis

In version `v2.8.0`, the resource `azapi_resource` was enhanced to support identity-based imports via the introduction of `IdentitySchema` ([internal/services/azapi_resource.go](https://github.com/Azure/terraform-provider-azapi/blob/main/internal/services/azapi_resource.go)). Under this schema, both `id` and `type` attributes are marked as `RequiredForImport` / `Required`:

```go
func (r *AzapiResource) IdentitySchema(ctx context.Context, request resource.IdentitySchemaRequest, response *resource.IdentitySchemaResponse) {
	response.IdentitySchema = identityschema.Schema{
		Attributes: map[string]identityschema.Attribute{
			"id": identityschema.StringAttribute{
				RequiredForImport: true,
				Description:       "The Azure resource ID.",
			},
			"type": identityschema.StringAttribute{
				RequiredForImport: true,
				Description:       "The Azure resource type.",
			},
		},
		Version: 0,
	}
}
```

Because `azapi_resource` implements `resource.ResourceWithMoveState`, the `terraform-plugin-framework` checks the schema of resource identity properties when `MoveResourceState` is executed. 

The framework expects the provider's `MoveState` function implementation to populate the `TargetIdentity` in the `MoveStateResponse` if an identity schema is defined. 

However, in `azapi`'s `MoveState` implementation, only `TargetState` and `TargetPrivate` are updated; `TargetIdentity` is completely omitted. Because of this, the framework detects empty `id` and `type` values on `TargetIdentity` and reports:

```text
Error: attributes "id" and "type" are required
```

---

### Debug Logs (TF_LOG=DEBUG)

Below are relevant log excerpts illustrating the framework discarding private states and entering the failure state:

```text
[WARN]  provider.terraform-provider-azapi_v2.10.0: Discarding incompatible resource private state data: error="json: cannot unmarshal object into Go value of type []uint8" tf_req_id=... tf_rpc=MoveResourceState @module=sdk.framework
Refreshing state... [id=/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg-myrg-001]
[ERROR] vertex "module.amba_alz[0].module.resource_group.azapi_resource.this" error: attributes "id" and "type" are required
[ERROR] vertex "module.amba_alz.module.resource_group.azapi_resource.this (expand)" error: attributes "id" and "type" are required
...
Error: attributes "id" and "type" are required
```

---

### Regression Summary
* **v2.7.0**: Works successfully (no `IdentitySchema` checks during `MoveState`).
* **v2.8.0 - v2.11.0**: Fails with `Error: attributes "id" and "type" are required`.

---

### Minimal Reproducible Example

Place these files in a clean directory:

`main.tf`:
```hcl
provider "azurerm" {
  features {}
}

provider "azapi" {}

resource "azapi_resource" "this" {
  location  = "westeurope"
  name      = "example-rg"
  parent_id = "/subscriptions/00000000-0000-0000-0000-000000000000"
  type      = "Microsoft.Resources/resourceGroups@2025-04-01"
}

moved {
  from = azurerm_resource_group.this
  to   = azapi_resource.this
}
```

If you have a pre-existing `azurerm_resource_group.this` in your state, running a plan using provider versions `2.8.0` through `2.11.0` will halt with the required attributes error during the schema-matching/state-moving routine.

---

### Expected Behavior

State moving should complete successfully, as it did in `v2.7.0`.

### Actual Behavior

Plan execution halts with `Error: attributes "id" and "type" are required`.

---

### Temporary Workaround

Pinning the `azapi` provider constraint strictly below `2.8.0` allows the migration to proceed safely:

```hcl
terraform {
  required_providers {
    azapi = {
      source  = "azure/azapi"
      version = "2.7.0"
    }
  }
}
```
---
Co-Authored by GitHub CoPilot

--- a/internal/services/azapi_resource.go
+++ b/internal/services/azapi_resource.go
@@ -1402,6 +1402,12 @@ func (r *AzapiResource) MoveState(ctx context.Context) []resource.StateMover {
 
 				response.Diagnostics.Append(response.TargetPrivate.SetKey(ctx, FlagMoveState, []byte("true"))...)
 				response.Diagnostics.Append(response.TargetState.Set(ctx, state)...)
+
+				resourceIdentity := AzapiResourceIdentityModel{
+					ID:   state.ID,
+					Type: state.Type,
+				}
+				response.Diagnostics.Append(response.TargetIdentity.Set(ctx, resourceIdentity)...)
 			},
 		},
 	}

--- a/internal/services/azapi_resource_identity_test.go
+++ b/internal/services/azapi_resource_identity_test.go
@@ -102,6 +102,82 @@ func TestAzapiResourceReadSetsIdentityWhenArmReturnsNotFound(t *testing.T) {
 	}
 }
 
+func TestAzapiResourceMoveStateSetsTargetIdentityForAzurermResources(t *testing.T) {
+	ctx := context.Background()
+	sourceID := "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/acctest-rg"
+
+	azapiResource := &services.AzapiResource{}
+	stateMovers := azapiResource.MoveState(ctx)
+	if len(stateMovers) == 0 {
+		t.Fatal("expected at least one state mover")
+	}
+
+	stateMover := stateMovers[0]
+
+	sourceState := tfsdk.State{Schema: *stateMover.SourceSchema}
+	diags := sourceState.Set(ctx, azurermResourceGroupMoveState{
+		ID:                    types.StringValue(sourceID),
+		ResourceManagerID:     types.StringNull(),
+		ResourceID:            types.StringNull(),
+		ResourceVersionlessID: types.StringNull(),
+	})
+	if diags.HasError() {
+		t.Fatalf("setting source state: %+v", diags)
+	}
+
+	var schemaResponse resource.SchemaResponse
+	azapiResource.Schema(ctx, resource.SchemaRequest{}, &schemaResponse)
+	if schemaResponse.Diagnostics.HasError() {
+		t.Fatalf("reading azapi schema: %+v", schemaResponse.Diagnostics)
+	}
+
+	var identitySchemaResponse resource.IdentitySchemaResponse
+	azapiResource.IdentitySchema(ctx, resource.IdentitySchemaRequest{}, &identitySchemaResponse)
+	if identitySchemaResponse.Diagnostics.HasError() {
+		t.Fatalf("reading azapi identity schema: %+v", identitySchemaResponse.Diagnostics)
+	}
+
+	moveRequest := resource.MoveStateRequest{
+		SourceTypeName: "azurerm_resource_group",
+		SourceState:    &sourceState,
+	}
+
+	moveResponse := resource.MoveStateResponse{
+		TargetState: tfsdk.State{
+			Schema: schemaResponse.Schema,
+		},
+		TargetIdentity: &tfsdk.ResourceIdentity{
+			Schema: identitySchemaResponse.IdentitySchema,
+			Raw:    tftypes.NewValue(identitySchemaResponse.IdentitySchema.Type().TerraformType(ctx), nil),
+		},
+	}
+	setEmptyMoveStateTargetPrivate(&moveResponse)
+
+	stateMover.StateMover(ctx, moveRequest, &moveResponse)
+	if moveResponse.Diagnostics.HasError() {
+		t.Fatalf("moving state: %+v", moveResponse.Diagnostics)
+	}
+
+	var movedState services.AzapiResourceModel
+	diags = moveResponse.TargetState.Get(ctx, &movedState)
+	if diags.HasError() {
+		t.Fatalf("reading moved state: %+v", diags)
+	}
+
+	var movedIdentity services.AzapiResourceIdentityModel
+	diags = moveResponse.TargetIdentity.Get(ctx, &movedIdentity)
+	if diags.HasError() {
+		t.Fatalf("reading moved identity: %+v", diags)
+	}
+
+	if movedIdentity.ID.ValueString() != movedState.ID.ValueString() {
+		t.Fatalf("expected moved identity ID %q, got %q", movedState.ID.ValueString(), movedIdentity.ID.ValueString())
+	}
+	if movedIdentity.Type.ValueString() != movedState.Type.ValueString() {
+		t.Fatalf("expected moved identity type %q, got %q", movedState.Type.ValueString(), movedIdentity.Type.ValueString())
+	}
+}
+
 type staticTokenCredential struct{}
 
 func (staticTokenCredential) GetToken(context.Context, policy.TokenRequestOptions) (azcore.AccessToken, error) {
@@ -130,4 +206,16 @@ func (t staticTransport) Do(req *http.Request) (*http.Response, error) {
 func setEmptyPrivateState(target any) {
 	field := reflect.ValueOf(target).Elem().FieldByName("Private")
 	field.Set(reflect.New(field.Type().Elem()))
+}
+
+func setEmptyMoveStateTargetPrivate(target any) {
+	field := reflect.ValueOf(target).Elem().FieldByName("TargetPrivate")
+	field.Set(reflect.New(field.Type().Elem()))
+}
+
+type azurermResourceGroupMoveState struct {
+	ID                    types.String `tfsdk:"id"`
+	ResourceManagerID     types.String `tfsdk:"resource_manager_id"`
+	ResourceID            types.String `tfsdk:"resource_id"`
+	ResourceVersionlessID types.String `tfsdk:"resource_versionless_id"`
 }

